services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -2586,11 +2586,23 @@ private void addSupersetSchemaForStore(String clusterName, String storeName, boo
 
   private void addWriteComputeSchemaForStore(String clusterName, String storeName, boolean dryRun) {
     Collection<SchemaEntry> valueSchemaEntries = getValueSchemas(clusterName, storeName);
+
     List<SchemaEntry> writeComputeSchemaEntries = new ArrayList<>(valueSchemaEntries.size());
+    int maxId = valueSchemaEntries.stream().map(SchemaEntry::getId).max(Comparator.naturalOrder()).get();
+
     for (SchemaEntry valueSchemaEntry: valueSchemaEntries) {
-      Schema writeComputeSchema =
-          writeComputeSchemaConverter.convertFromValueRecordSchema(valueSchemaEntry.getSchema());
-      writeComputeSchemaEntries.add(new SchemaEntry(valueSchemaEntry.getId(), writeComputeSchema));
+      Schema writeComputeSchema = null;
+      try {
+        writeComputeSchema = writeComputeSchemaConverter.convertFromValueRecordSchema(valueSchemaEntry.getSchema());
+      } catch (VeniceException e) {
```

Comment:
Hmm, I can understand your overall intention is to avoid bad schema avoiding good schema registration.
However, I am afraid that user can still try to use old "bad" schema to compose updates (is this possible?) And during deserialization, server won't find the writer schema.

Or even if the above case works, there is another case that, user will use the SchemaFetcher to fetch the partial update schema with a hardcoded bad value schema and it will fail since it does not have the corresponding partial update schema...

Reply:
Even this works, can it pass superset schema compatibility check logic?